### PR TITLE
Get rid of CalibratedKymographChannel

### PR DESCRIPTION
**Pull request by rpauszek**
**Why this PR?**

Previously we needed an intermediate utility class between `Kymo` and the kymotracking algorithms (specifically to handle downsampled kymos). However, with the current structure this class has become just an extra layer with no real purpose. So let's get rid of it.

**What to look for:**
- I started by going through each module that depended on `CalibratedKymographChannel` and removed all dependencies
- Added some convenience features to `KymoLine` in order to simplify some code; I kept them public as it seems `line_time_seconds` and `pixelsize` are useful features
- `CalibratedKymographChannel._to_pixel_rect()` was only used to set the ROI in the kymotracker widget, so I made this a standalone function there; in the future we may want to think about having a dedicated ROI class for this kind of stuff (along with widefield)
- None of the assertions were touched in the tests; the only changes are creating the test data, primarily switching from `CalibratedKymographChannel` to `Kymo`
- Removed the pixel/physical conversion methods (ie, `from_position`, `to_seconds`, etc...) as these are one-liners and only used in one or two places
- Changed the calculation of the `extent` argument for `Kymo` plotting. Now it is the same as what `CalibratedKymographChannel` had and defines the pixel position at the center (same as we do with the time axis)

### CODE REVIEW

**Review by JoepVanlier — CHANGES_REQUESTED** [review]
Great work! It's great that we've gotten rid of that wart.

One thing to think about for a future PR would be to have a more central place for these coordinate transforms. I'm a little concerned about having many places with these same linear transforms (between (rounded) pixels and coordinates/seconds). Without a way to easily convert between them in a central location it seems likely that we end up spreading transformations around.

Please don't forget to add a changelog entry!

File: lumicks/pylake/nb_widgets/kymotracker_widgets.py [hunk 1]
```diff
@@ -570,12 +566,12 @@ def __init__(
             "vel".
         """
         algorithm = track_greedy
-        calibrated_kymo_channel = CalibratedKymographChannel.from_kymo(kymo, channel)
-        position_scale = calibrated_kymo_channel._pixel_size
+        _data = kymo.get_image(channel)
```
[the remark this answers is not in the document]
Comment (rpauszek) [reply lumicks/pylake/nb_widgets/kymotracker_widgets.py]: temporary insanity?

File: lumicks/pylake/kymotracker/kymoline.py [hunk 2]
```diff
@@ -163,26 +163,34 @@ def seconds(self):
     def position(self):
         return self._localization.position
 
+    @property
+    def line_time_seconds(self):
```
[the remark this answers is not in the document]
Comment (rpauszek) [reply lumicks/pylake/kymotracker/kymoline.py]: sure sounds good. I just made them both private, we can always change it later

File: lumicks/pylake/nb_widgets/kymotracker_widgets.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (JoepVanlier) [reply lumicks/pylake/nb_widgets/kymotracker_widgets.py]: 😆

File: lumicks/pylake/kymotracker/kymotracker.py [hunk 3]
```diff
@@ -23,15 +23,22 @@
...
+    pixelsize: float
+        Size of spatial pixel dimension in physical units.
+    line_time_seconds: float
+        Size of temporal pixel dimension in seconds.
+    """
     return [
-        [from_seconds(t0), from_position(p0)],
-        [from_seconds(t1), from_position(p1)],
+        [int(seconds / line_time_seconds), int(position / pixelsize)]
+        for (seconds, position) in rect
```
[the remark this answers is not in the document]
Comment (JoepVanlier) [reply lumicks/pylake/kymotracker/kymotracker.py:41]: Decided to keep it as it is.

File: lumicks/pylake/kymotracker/tests/test_kymoline.py [hunk 4]
```diff
@@ -2,80 +2,76 @@
...
-    k1 = KymoLine([1, 2, 3], [1, 2, 3], channel)
+    k1 = KymoLine([1, 2, 3], [1, 2, 3], blank_kymo, "red")
     k2 = k1.with_offset(2, 2)
     assert id(k2) != id(k1)
 
     np.testing.assert_allclose(k2.coordinate_idx, [3, 4, 5])
-    assert k1._image == channel
-    assert k2._image == channel
-
+    assert k1._kymo == blank_kymo
```
[the remark this answers is not in the document]
Comment (JoepVanlier) [reply lumicks/pylake/kymotracker/tests/test_kymoline.py:32]: 👍

**Review by JoepVanlier — APPROVED** [review]
:shipit:

**Pull request merged**